UnityProject/Assets/Scripts/Weapons/Projectiles/Behaviours/KeneticProjectileDamageInterity.cs

```diff
@@ -0,0 +1,55 @@
+﻿using UnityEngine;
+using ScriptableObjects.Gun;
+
+namespace Weapons.Projectiles.Behaviours
+{
+	/// <summary>
+	/// Damages integrity on collision
+	/// only for kenetic weapons
+	/// </summary>
+	public class KeneticProjectileDamageInterity : MonoBehaviour, IOnShoot, IOnHit
+	{
+		private GameObject shooter;
+		private Gun weapon;
+
+		[SerializeField] private DamageData damageData = null;
+
+		public void OnShoot(Vector2 direction, GameObject shooter, Gun weapon, BodyPartType targetZone = BodyPartType.Chest)
+		{
+			this.shooter = shooter;
+			this.weapon = weapon;
+		}
+
+		public bool OnHit(RaycastHit2D hit)
+		{
+			return TryDamage(hit);
+		}
+
+		private bool TryDamage(RaycastHit2D hit)
+		{
+			float pressure = MatrixManager.AtPoint((Vector3Int)hit.point.To2Int(), true).MetaDataLayer.Get(hit.transform.localPosition.RoundToInt()).GasMix.Pressure;
+			var newDamage = damageData.Damage;
+			var coll = hit.collider;
+			var integrity = coll.GetComponent<Integrity>();
+			if (integrity == null) return false;
+			// checks if its a high atmosphere 
+
+			newDamage = (1175 / 26) - ((15 - pressure) / 26);
```

Comment:
there is no point in line `30` if you are reassigning `newDamage` here. So you can get rid of line `30` and turn line `37` into `var newDamage = (1175 / 26) - ((15 - pressure) / 26);`